gapis/api/vulkan/frame_loop.go

```diff
@@ -779,14 +791,75 @@ func (f *frameLoop) buildStartEndStates(ctx context.Context, startState *api.Glo
 				f.renderPassToCreate[renderPass] = true
 			}
 
-			// TODO:  Recreate destroyed resources.
-		}
+		// Command Buffers
+		case *VkAllocateCommandBuffers:
+			vkCmd := cmd.(*VkAllocateCommandBuffers)
+			cmdBuffers := vkCmd.PCommandBuffers().MustRead(ctx, vkCmd, currentState, nil)
+			f.commandBufferToFree[cmdBuffers] = true
+			log.D(ctx, "Command buffer %v allcoated.", cmdBuffers)
+		case *VkFreeCommandBuffers:
+			vkCmd := cmd.(*VkFreeCommandBuffers)
+			cmdBufCount := vkCmd.CommandBufferCount()
+			cmdBufs := vkCmd.PCommandBuffers().Slice(0, uint64(cmdBufCount), currentState.MemoryLayout).MustRead(ctx, cmd, currentState, nil)
+			for _, cmdBuf := range cmdBufs {
+				// The command buffer deleted in this call was created during loop, no action needed.
+				log.D(ctx, "Command buffer %v freed", cmdBufs)
+				if _, ok := f.commandBufferToFree[cmdBuf]; ok {
+					delete(f.commandBufferToFree, cmdBuf)
+				} else { // The command buffer deleted in this call was not created during loop, need to back up it
+					f.commandBufferToAllocate[cmdBuf] = true
+				}
+			}
+		case *VkBeginCommandBuffer:
```

Comment:
What if the begin was just before the loop start? Under this scheme, do we not need to consider VkEndCommandBuffer and VkCmd* as editing commands too?

I was talking to Andrew and I think we need to reconstruct every CommandBuffer submitted between the start and end of a loop that was *not* created inside the loop body itself. This will obviously touch more command buffers than strictly necessary in some situations, but it's the safest simplest option for now. We can optimise it to be less conservative over time if it turns out to be a performance issue.

Reply:
That would also let us get rid of the complexity of the "commandBufferNotRecorded" map.

Reply:
sure, we can do that and that is simpler to implement.